### PR TITLE
Support pause and resume

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

**Comment by ansyral** [comment]
:shipit:

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java [hunk 1]
```diff
@@ -466,13 +472,13 @@ private void handleEvent(DebugEvent debugEvent) {
...
-            try {
-                Events.StoppedEvent stopEvent = this.convertDebuggerStoppedEventToClient((LocatableEvent) event);
-                this.sendEvent(stopEvent);
-            } catch (AbsentInformationException | URISyntaxException e) {
-                Logger.logException("Get breakpoint/step info exception", e);
-            }
+        } else if (event instanceof BreakpointEvent) {
+            ThreadReference bpThread = ((BreakpointEvent) event).thread();
+            this.sendEvent(new Events.StoppedEvent("breakpoint", bpThread.uniqueID()));
+            debugEvent.shouldResume &= false;
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java]: Why do we use &= operator? EventHub actually does so.
I think the assignment is enough here.

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java [hunk 2]
```diff
@@ -466,13 +472,13 @@ private void handleEvent(DebugEvent debugEvent) {
...
+        } else if (event instanceof StepEvent) {
+            ThreadReference stepThread = ((StepEvent) event).thread();
+            this.sendEvent(new Events.StoppedEvent("step", stepThread.uniqueID()));
             debugEvent.shouldResume &= false;
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java]: Same here. The assignment operator is enough.

**Review by akaroml — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java (hunk 1 above)
Comment (testforstephen) [reply org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java]: agree.

**Review by akaroml — APPROVED** [review]
Nice job! Thanks.

**Pull request merged**